nexus/src/db/datastore.rs

```diff
@@ -1750,16 +1750,16 @@ impl DataStore {
         &self,
         opctx: &OpContext,
         authz_instance: &authz::Instance,
-        pagparams: &DataPageParams<'_, Name>,
     ) -> ListResultVec<NetworkInterface> {
         opctx.authorize(authz::Action::ListChildren, authz_instance).await?;
-
         use db::schema::network_interface::dsl;
-        paginated(dsl::network_interface, dsl::name, &pagparams)
+
+        dsl::network_interface
             .filter(dsl::time_deleted.is_null())
-            .filter(dsl::instance_id.eq(authz_instance.id()))
+            .filter(dsl::vpc_id.eq(authz_instance.id()))
+            .order(dsl::name.asc())
```

Comment:
Let's order these ascending by `dsl::slot`. The NIC in slot zero is always the primary interface, and so the external face of the instance on the network. Other than that they have no intrinsic ordering.